src/components/form/FormFieldVoucher.vue

```diff
@@ -0,0 +1,158 @@
+<script lang="ts">
+/**
+ * FormFieldVoucher Component
+ *
+ * @description * Use this component to render a Voucher widget.
+ * It handles voucher submission and validation and emits
+ * the data about the voucher.
+ *
+ * Used in `RegisterChallengePayment` component.
+ *
+ * @events
+ * - `update:voucher`: Emitted after voucher is successfully applied.
+ *
+ * @components
+ * - `FormFieldTextRequired`: Component to render required text field.
+ *
+ * @example
+ * <form-field-voucher @update:voucher="onUpdateVoucher" />
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=6410-2305&t=gB7ERmDZorpD4TdE-1)
+ */
+
+// libraries
+import { Notify } from 'quasar';
+import { defineComponent, ref } from 'vue';
+
+// components
+import FormFieldTextRequired from '../global/FormFieldTextRequired.vue';
+
+// composables
+import { i18n } from '../../boot/i18n';
+
+// types
+import type { FormPaymentVoucher } from '../types/Form';
+
+// fixtures
+import voucherFull from '../../../test/cypress/fixtures/registerPaymentVoucherFull.json';
+import voucherHalf from '../../../test/cypress/fixtures/registerPaymentVoucherHalf.json';
+
+export default defineComponent({
+  name: 'FormFieldVoucher',
+  components: {
+    FormFieldTextRequired,
+  },
+  emits: ['remove:voucher', 'update:voucher'],
+  setup(props, { emit }) {
+    const code = ref('');
+    const voucher = ref<FormPaymentVoucher | null>(null);
+
+    /**
+     * Submits voucher data to API
+     * If voucher is valid it emits the data
+     * @returns {void}
+     */
+    const onSubmitVoucher = (): void => {
+      // TODO: Add API call and remove dummy data
+      if (code.value === 'FULL') {
+        voucher.value = voucherFull;
+      }
+      if (code.value === 'HALF') {
+        voucher.value = voucherHalf;
+      }
+      if (voucher.value) {
+        Notify.create({
+          type: 'positive',
+          message: i18n.global.t('notify.voucherApplySuccess'),
+        });
+        emit('update:voucher', voucher.value);
+      } else {
+        Notify.create({
+          type: 'negative',
+          message: i18n.global.t('notify.voucherApplyError'),
```

Comment:
Better position top middle, and you can define global config variable for notification message position:

```ts
const notifyMessagePosition =  rideToWorkByBikeConfig.notifyMessagePosition;

Notify.create({
  ...,
  position: notifyMessagePosition,
});
```

Reply:
The position is defined globally in `quasar.config.js`.
I updated the value and added the same config for the component tests in `component.js` as well.
(Or do you want to use a global config values in these two config files? Import of config into `quasar.config.js` did not seem to work.

<img width="643" alt="Screenshot 2024-08-26 at 17 04 21" src="https://github.com/user-attachments/assets/970dde8f-7d4a-48e3-8478-f11b6e0db858">

Reply:
> The position is defined globally in `quasar.config.js`. I updated the value and added the same config for the component tests in `component.js` as well. (Or do you want to use a global config values in these two config files? 

It will be better to use global config var, please.

./ride_to_work_by_bike_config.toml

```toml
notifyMessagePosition = "top"
```
./src/utils/get_app_conf.js

```js
...
  } else if (process.env.NOTIFY_MESSAGE_POSITION) {
    config['notifyMessagePosition'] = process.env.NOTIFY_MESSAGE_POSITION;
  }
```

> Import of config into `quasar.config.js` did not seem to work.

Works as expected. See code line no. 64, please.


./quasar.config.js

```js
framework: {
      config: {
        notify: {
          position: getAppConfig(process)['notifyMessagePosition'],
        },
      },
...
},
```

Reply:
Thank you. Updated from config.